### ISSUE TITLE
condition block arguments for oci_network_firewall_network_firewall_policy_security_rule are optional, but required by resource

### ISSUE THREAD

**Issue by rlopezl** [oracle/terraform-provider-oci#2005]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Affected Resource(s)

oci_network_firewall_network_firewall_policy_security_rule

### Terraform Configuration Files

main.tf

```hcl

locals {
  prefix = join("-", tolist(["or", lower(var.tags.region), lower(var.tags.enterprise), lower(var.tags.account), lower(var.tags.system), ""]))

  _rules         = var.firewall_policy_security_rules != null ? var.firewall_policy_security_rules : {}
  _address_lists = var.firewall_policy_address_lists == null ? {} : var.firewall_policy_address_lists

  rules = { for k, v in local._rules : k => v }
  address_lists = {
    source      = { for k, v in local._address_lists : k => v if v.kind == "source" }
    destination = { for k, v in local._address_lists : k => v if v.kind == "destination" }
  }
}

resource "oci_network_firewall_network_firewall_policy" "policy" {
  compartment_id = var.compartment_id

  display_name = format("${local.prefix}${lower(var.firewall_policy_name)}-%s-firewallpolicy%.2d-%s", var.tags.security_exposure_level, var.offset,
    var.tags.environment
  )

  freeform_tags = merge(
    var.tags,
    {
      "resource_name" = format("${local.prefix}${lower(var.firewall_policy_name)}-%s-firewallpolicy%.2d-%s", var.tags.security_exposure_level, var.offset,
        var.tags.environment
      )
      "deployment_date" = formatdate("DD-MM-YYYY", timestamp()),
      "resource_type"   = "firewall-policy",
    }
  )
}

resource "oci_network_firewall_network_firewall_policy_address_list" "source" {
  for_each = local.address_lists.source

  name = lower(each.value.name)

  network_firewall_policy_id = oci_network_firewall_network_firewall_policy.policy.id
  type                       = each.value.type

  addresses = each.value.addresses
}

resource "oci_network_firewall_network_firewall_policy_address_list" "destination" {
  for_each = local.address_lists.destination

  name = lower(each.value.name)

  network_firewall_policy_id = oci_network_firewall_network_firewall_policy.policy.id
  type                       = each.value.type

  addresses = each.value.addresses
}

resource "oci_network_firewall_network_firewall_policy_security_rule" "security_rules" {
  for_each                   = local.rules
  network_firewall_policy_id = oci_network_firewall_network_firewall_policy.policy.id

  action = each.value.action

  dynamic "condition" {
    for_each = lookup(local.rules, "condition", true) ? [{
      application         = lookup(each.value.condition, "application", null)
      destination_address = lookup(each.value.condition, "destination_address", null)
      service             = lookup(each.value.condition, "service", null)
      source_address      = lookup(each.value.condition, "source_address", null)
      url                 = lookup(each.value.condition, "url", null)
    }] : []
    content {
      application         = condition.value.application
      destination_address = condition.value.destination_address != [""] ? [for key, value in local.address_lists.destination : oci_network_firewall_network_firewall_policy_address_list.destination[key].name] : null
      service             = condition.value.service
      source_address      = condition.value.source_address != [""] ? [for key, value in local.address_lists.source : oci_network_firewall_network_firewall_policy_address_list.source[key].name] : null
      url                 = condition.value.url
    }
  }

  name = format("${local.prefix}${lower(each.value.name)}-%s-policyrule%.2d-%s", var.tags.security_exposure_level, var.offset, var.tags.environment)

  inspection = each.value.action == "INSPECT" ? each.value.inspection : null

  depends_on = [oci_network_firewall_network_firewall_policy_address_list.destination, oci_network_firewall_network_firewall_policy_address_list.source]
}

``` 
variables.tf

```hcl

variable "compartment_id" {
  type        = string
  description = "The OCID of the compartment to contain the VCN."
}

variable "offset" {
  type    = number
  default = 1
}

variable "firewall_policy_name" {
  type        = string
  description = "A user-friendly name. Does not have to be unique, and it's changeable. Avoid entering confidential information."
}

variable "firewall_policy_address_lists" {
  description = <<EOH
  List of Address List to create for network firewall.
  <a name=kind:></a>[kind:](#kind:) Kind of address List. The accepted values are - * destination * source.
  <a name=addresses:></a>[action:](#action:) Unique name to identify the group of addresses to be used in the policy rules.
  <a name=name:></a>[name:](#name:) Name for the Security rule, must be unique within the policy.
  <a name=type:></a>[type:](#type:) Type of address List. The accepted values are - * FQDN * IP.
  EOH
  type = map(object({
    kind      = string
    addresses = list(string)
    name      = string
    type      = string
  }))
  default = null
}

variable "firewall_policy_security_rules" {
  description = <<EOH
  List of Security Rules defining the behavior of the policy. The first rule with a matching condition determines the action taken upon network traffic.
  <a name=action:></a>[action:](#action:) Types of Action on the Traffic flow.
  <a name=condition:></a>[condition:](#condition) Criteria to evaluate against network traffic. A match occurs when at least one item in the array associated with each specified property corresponds with the relevant aspect of the traffic.
    <ul>
      <li><a name=applications:></a>[applications:](#applications:) (Optional) (Updatable) An array of application list names to be evaluated against the traffic protocol and protocol-specific parameters.</li>
      <li><a name=destinations:></a>[destinations:](#destinations:) (Optional) (Updatable) An array of IP address list names to be evaluated against the traffic destination address.</li>
      <li><a name=sources:></a>[sources:](#sources:) (Optional) (Updatable) An array of IP address list names to be evaluated against the traffic source address.</li>
      <li><a name=urls:></a>[urls:](#urls:) An array of URL pattern list names to be evaluated against the HTTP(S) request target.</li>
    </ul>
  <a name=inspection:></a>[inspection:](#inspection:) Type of inspection to affect the Traffic flow. This is only applicable if action is INSPECT.
  <a name=position:></a>[position:](#position) An object which defines the position of the rule. Only one of the following position references should be provided.
    <ul>
      <li><a name=after_rule:></a>[after_rule:](#after_rule:) (Optional) (Updatable) Identifier for rule after which this rule lies.</li>
      <li><a name=before_rule:></a>[before_rule:](#before_rule:) (Optional) (Updatable) Identifier for rule before which this rule lies.</li>
    </ul>
  <a name=name:></a>[name:](#name:) Name for the Security rule, must be unique within the policy.
  EOH
  type = map(object({
    action = string
    condition = object({
      application         = optional(list(string))
      destination_address = optional(list(string))
      service             = optional(list(string))
      source_address      = optional(list(string))
      url                 = optional(list(string))
    })
    inspection = optional(string)
    position = optional(object({
      after_rule  = optional(string)
      before_rule = optional(string)
    }))
    name = string
  }))
  default = {
    "1" = {
      action    = ""
      condition = {}
      name      = ""
    }
  }
}

variable "tags" {
  description = "A mapping of tags to assign to role definition"
  type        = map(string)
}

```
terraform.tfvars

```hcl
compartment_id = "ocid1.compartment.oc1..aaaaaaaah65wvkxzhqq777ingcvpzlighuom44mvreqqg25pziq7xldoc4ma"

# Firewall Policy vars
firewall_policy_name = "homsv"
firewall_policy_address_lists = {
  "1" = {
    kind      = "destination"
    name      = "homsv-dst"
    addresses = ["11.100.0.1/32"]
    type      = "IP"
  },
  "2" = {
    kind      = "source"
    name      = "homsv-src"
    addresses = ["10.0.1.16/32"]
    type      = "IP"
  }
}
firewall_policy_security_rules = {
  "1" = {
    name   = "homsv"
    action = "ALLOW"
    condition = {
      application = ["null"]
      service     = ["null"]
    }
  }
}
```

### Debug Output

https://gist.github.com/rlopezl/b0f4b024dfb4e5aa4d6055caf9367a2f

### Expected Behavior

The variables of condition must be optional, as the documentation said (https://registry.terraform.io/providers/oracle/oci/latest/docs/resources/network_firewall_network_firewall_policy_security_rule):

![Captura desde 2023-12-12 12-02-39](https://github.com/oracle/terraform-provider-oci/assets/28804062/b80ad27f-0609-4134-9a8e-3d763e47a368)

### Actual Behavior

The condition arguments, that are optional, are required for the module but in OCI console is optional too:

```
│ Error: Missing required argument
│ 
│   with module.firewall.module.firewall_policy.oci_network_firewall_network_firewall_policy_security_rule.security_rules["1"],
│   on ../../modules/firewall_policy/main.tf line 55, in resource "oci_network_firewall_network_firewall_policy_security_rule" "security_rules":
│   55: resource "oci_network_firewall_network_firewall_policy_security_rule" "security_rules" {
│ 
│ The argument "condition.0.url" is required, but no definition was found.
```

### Steps to Reproduce

1. `terraform apply`

### References

https://registry.terraform.io/providers/oracle/oci/latest/docs/resources/network_firewall_network_firewall_policy_security_rule
https://docs.oracle.com/en-us/iaas/Content/network-firewall/security-rule-create.htm

**Comment by tf-oci-pub** [comment]
Thank you for reporting the issue. We have raised an internal ticket to track this. Our service engineers will get back to you.

**Comment by vsin12** [comment]
Resolved in v5.25.0
